presto-main/src/main/java/io/prestosql/execution/scheduler/UniformNodeSelectorFactory.java

```diff
@@ -75,16 +77,33 @@ public UniformNodeSelectorFactory(
         checkArgument(maxSplitsPerNode >= maxPendingSplitsPerTask, "maxSplitsPerNode must be > maxPendingSplitsPerTask");
     }
 
+    @VisibleForTesting
+    public UniformNodeSelectorFactory(
+            InternalNodeManager nodeManager,
+            NodeSchedulerConfig config,
+            NodeTaskMap nodeTaskMap,
+            boolean memoizeNodeMap)
```

Comment:
1. Let's make this method initialize all fields and the one that has `@Inject` call `this(...)`.
2. Let's make this method accept memoization timeout (with `0` meaning memoization disabled)